### PR TITLE
Fix NumberFormatInfoNumberGroupSizes test for certain OS

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes https://github.com/dotnet/corefx/issues/17713

On Windows 7 the culture is not found. I thought it was excluded because the expected result was different. Reverting the condition back to how it was.

@tarekgh

### CODE REVIEW

**Comment by danmosemsft** [comment]
Also fix https://github.com/dotnet/corefx/issues/17643 'again'.
Before my change, the test was not running on ur-IN at all on these platforms, so I had no way to know what the expected values were. 
The CI did not run Fedora 24, Ubuntu 1604 or 1610.

**Comment by danmosemsft** [comment]
@dotnet-bot help please

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop Fedora24 Debug please
@dotnet-bot test innerloop Ubuntu16.10 Debug please
@dotnet-bot test innerloop Ubuntu16.04 Release please
@dotnet-bot test outerloop netcoreapp Windows 7 Debug please

File: src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs [hunk 1]
```diff
@@ -9,13 +9,12 @@ internal static class NumberFormatInfoData
...
             if (
-                ((PlatformDetection.IsWindows && PlatformDetection.WindowsVersion >= 10)
+                (PlatformDetection.IsWindows && PlatformDetection.WindowsVersion >= 10)
 #if !uap
                 ||
                 (PlatformDetection.IsOSX && PlatformDetection.OSXKernelVersion >= new Version(15, 0))
 #endif
-                ) &&
-                (!PlatformDetection.IsUbuntu1510 && !PlatformDetection.IsUbuntu1604 && !PlatformDetection.IsUbuntu1610)
+                || PlatformDetection.IsUbuntu1510 || PlatformDetection.IsUbuntu1604 || PlatformDetection.IsUbuntu1610 || PlatformDetection.IsFedora
```
Comment (danmosemsft) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs]: @tarekgh I don't think Helix or Jenkins have runs for 1510 and i don't see it in our support matrix either. Just 1404, 1604, 1610. Perhaps we should remove IsUbuntu15010 everwhere.  @leecow ?

Comment (danmosemsft) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs]: (So i have no idea whether this is right for 1510 .. but I guess no tests will run so it doesn't matter.)

**Comment by danmosemsft** [comment]
@dotnet-bot help please

**Comment by danmosemsft** [comment]
@dotnet-bot test outerloop netcoreapp Windows 10 Debug

**Comment by tarekgh** [comment]
LGTM. thanks @danmosemsft

**Comment by danmosemsft** [comment]
@dotnet-bot test inner loop fedora24 debug please (networking test)

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop fedora24 debug please (networking test)

**Comment by danmosemsft** [comment]
I've opend https://github.com/dotnet/corefx/issues/17749 for the test. It's clearly not related. I'll talk to @Priya91 about whether to disable it. Meanwhile I'll merge.

**Pull request merged**